packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts

```diff
@@ -1,147 +1,289 @@
-    } else {
-      await prisma.booking.update({
-        where: {
-          id: bookingId,
-        },
-        data: {
-          location,
-          references: {
-            create: updatedResult.referencesToCreate,
-          },
-        },
-      });
-
-      const metadata: AdditionalInformation = {};
-      if (results.length) {
-        metadata.hangoutLink = results[0].updatedEvent?.hangoutLink;
-        metadata.conferenceData = results[0].updatedEvent?.conferenceData;
-        metadata.entryPoints = results[0].updatedEvent?.entryPoints;
-      }
-      try {
-        await sendLocationChangeEmails(
-          { ...evt, additionalInformation: metadata },
-          booking?.eventType?.metadata as EventTypeMetadata
-        );
-      } catch (error) {
-        console.log("Error sending LocationChangeEmails");
-      }
-    }
-  } catch {
-    throw new TRPCError({ code: "INTERNAL_SERVER_ERROR" });
+export function getLocationForOrganizerDefaultConferencingAppInEvtFormat({
+  organizer,
+  loggedInUserTranslate: translate,
+}: {
+  organizer: {
+    name: string;
+    metadata: {
+      defaultConferencingApp?: NonNullable<UserMetadata>["defaultConferencingApp"];
+    } | null;
+  };
+  /**
+   * translate is used to translate if any error is thrown
+   */
+  loggedInUserTranslate: Awaited<ReturnType<typeof getTranslation>>;
+}) {
+  const organizerMetadata = organizer.metadata;
+  const defaultConferencingApp = organizerMetadata?.defaultConferencingApp;
+  if (!defaultConferencingApp) {
+    throw new UserError(
+      translate("organizer_default_conferencing_app_not_found", { organizer: organizer.name })
+    );
+  }
+  const defaultConferencingAppSlug = defaultConferencingApp.appSlug;
+  const app = getAppFromSlug(defaultConferencingAppSlug);
+  if (!app) {
+    throw new SystemError(`Default conferencing app ${defaultConferencingAppSlug} not found`);
+  }
+  const defaultConferencingAppLocationType = app.appData?.location?.type;
+  if (!defaultConferencingAppLocationType) {
+    throw new SystemError("Default conferencing app has no location type");
   }
+
+  const location = defaultConferencingAppLocationType;
+  const locationType = getEventLocationType(location);
+  if (!locationType) {
+    throw new SystemError(`Location type not found: ${location}`);
+  }
+
+  if (locationType.linkType === "dynamic") {
+    // Dynamic location type need to return the location as it is e.g. integrations:zoom_video
+    return location;
+  }
+
+  const appLink = defaultConferencingApp.appLink;
+  if (!appLink) {
+    throw new SystemError(`Default conferencing app ${defaultConferencingAppSlug} has no app link`);
+  }
+  return appLink;
+}
+
+export async function editLocationHandler({ ctx, input }: EditLocationOptions) {
+  const { newLocation, credentialId: conferenceCredentialId } = input;
+  const { booking, user: loggedInUser } = ctx;
+
+  const organizer = await UserRepository.findByIdOrThrow({ id: booking.userId || 0 });
```

Comment:
When would the `booking.userId` not be defined? Should we instead throw an error?

Reply:
Yeah, certainly that is the best way.  Added a followup ticket